server/src/main/java/org/opensearch/index/seqno/ReplicationTracker.java

```diff
@@ -1252,7 +1252,7 @@ public ReplicationCheckpoint getLatestReplicationCheckpoint() {
     }
 
     private boolean isPrimaryRelocation(String allocationId) {
-        Optional<ShardRouting> shardRouting = routingTable.shards()
+        Optional<ShardRouting> shardRouting = routingTable.assignedShards()
```

Comment:
a nitpick, we can simplify this function a bit and still include the null check to be defensive...

```java
    private boolean isPrimaryRelocation(String allocationId) {
        return routingTable.assignedShards()
            .stream()
            .filter(routing -> Objects.nonNull(routing.allocationId()))
            .filter(routing -> routing.allocationId().getId().equals(allocationId))
            .anyMatch(ShardRouting::primary);
    }
```